### ISSUE TITLE
[Rule Request] no-promise-in-boolean-expressions

### ISSUE THREAD

**Issue by danielberndt** [palantir/tslint#3804]
We're working with a lot of async/await in our backend code, and the most frequent source of mistake I found in our code base has this shape:

```js
if (user.pictures.exists()) doX()
```

whereas it should be 

```js
if (await user.pictures.exists()) doX()
```

I'm aware of the `strict-boolean-expressions` rule, but its ruleset is too strict for our code. So I'm suggesting to create a softer version which allows anything but promises in boolean expressions.

**Comment by JoshuaKGoldberg** [comment]
> but its ruleset is too strict for our code

Specifically which parts? Maybe the rule could have options added that would allow it to be flexible enough for you?

**Comment by vaskevich** [comment]
+1 to the strictness of that rule. We have a lot of code that looks like this:

```
let maybeObj: SomeObject | undefined = getObj();

if (maybeObj) {
  // do something
}
```

If this logic was put into that rule, I'd want a rule like `"allow-all-unions-except": <symbols>` where `<symbols>` is an array of classes that cannot be in a boolean expression. To handles promises correctly, I'd set that to `["Promise", "Bluebird"]` in our codebase.

**Comment by vaskevich** [comment]
I did find something like this elsewhere: https://github.com/wwwouter/tslint-contrib but unfortunately it doesn't support blacklisting additional classes like `"Bluebird"`.

**Comment by JoshuaKGoldberg** [comment]
Looks like discussion continued a bit deeper in #3983 for a very similar idea. Closing this issue as duplicate; let's continue talking there.

**Comment by whitecolor** [comment]
Right, this is a common bug.
